File: 프로그래머스/lv2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        Integer[] citationss = Arrays.stream(citations).boxed().toArray(Integer[]::new); // int형 -> Integer형변환
        Arrays.sort(citationss, (a,b) -> b - a); // 배열 내림차순 정렬 시 int(x) Integer(o)
        for(int i = 0; i < citationss.length; i++){
            if((i + 1) > citationss[i]){
                return i;
            }
        }
        return answer == 0 ? citationss.length : answer;
    }
}

File: 프로그래머스/lv2/42747. H－Index/H－Index.js
function solution(citations) {
    let answer = 0;
    citations.sort((a, b) => b - a);
    
    for(let i = 0; i < citations.length; i++){
        if(i + 1 > citations[i]){
            return i;
        }
    }
    return answer == 0 ? citations.length : answer;
}